File: frontend/src/core/services/salary.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

import { environment } from '../../environments/environment';

import {
  SalaryResponse,
  UpdateSalaryRequest,
  SalaryHistoryResponse
} from '../models/salary.model';
import { PageResponse } from '../models/page-response.model';

@Injectable({
  providedIn: 'root'
})
export class SalaryService {

  private readonly http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiUrl}/employees`;

  getCurrentSalary(employeeId: number): Observable<SalaryResponse> {
    return this.http.get<SalaryResponse>(
      `${this.baseUrl}/${employeeId}/salary`
    );
  }

  updateSalary(
    employeeId: number,
    request: UpdateSalaryRequest
  ): Observable<SalaryResponse> {
    return this.http.put<SalaryResponse>(
      `${this.baseUrl}/${employeeId}/salary`,
      request
    );
  }

  getSalaryHistory(
    employeeId: number,
    page = 0,
    size = 20
  ): Observable<PageResponse<SalaryHistoryResponse>> {

    const params = new HttpParams()
      .set('page', page)
      .set('size', size);

    return this.http.get<PageResponse<SalaryHistoryResponse>>(
      `${this.baseUrl}/${employeeId}/salary-history`,
      { params }
    );
  }
}

File: backend/src/main/java/com/acme/salary/salary/service/SalaryService.java
package com.acme.salary.salary.service;

import com.acme.salary.salary.dto.SalaryHistoryResponse;
import com.acme.salary.salary.dto.SalaryResponse;
import com.acme.salary.salary.dto.UpdateSalaryRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface SalaryService {

    SalaryResponse getCurrentSalary(Long employeeId);

    SalaryResponse updateSalary(
            Long employeeId,
            UpdateSalaryRequest request
    );

    Page<SalaryHistoryResponse> getSalaryHistory(
            Long employeeId,
            Pageable pageable
    );
}